### ISSUE TITLE
Redundant calls to "describe_stacks" from Aws::CloudFormation::Stack#data

### ISSUE THREAD

**Issue by mdub** [aws/aws-sdk-ruby#1265]
Most resource-collection methods appear to fully "load" with a single API call:

```
$ AWS_DEBUG=true aws.rb
Aws> Aws.ec2.resource.instances.first.data; nil
[Aws::EC2::Client 200 1.41273 0 retries] describe_instances()  
=> nil
Aws> Aws.ec2.resource.subnets.first.data; nil
[Aws::EC2::Client 200 0.250955 0 retries] describe_subnets()  
=> nil
Aws> Aws.ec2.resource.security_groups.first.data; nil
[Aws::EC2::Client 200 0.760407 0 retries] describe_security_groups()  
=> nil
Aws> Aws.ec2.resource.volumes.first.data; nil
[Aws::EC2::Client 200 1.107822 0 retries] describe_volumes()  
=> nil
```

but calling `Aws::CloudFormation::Stack#data` appears to trigger another call to `describe_stacks`, which I imagine to be redundant:

```
Aws> Aws.cloudformation.resource.stacks.first.data; nil
[Aws::CloudFormation::Client 200 0.949843 0 retries] describe_stacks()  
[Aws::CloudFormation::Client 200 0.05216 0 retries] describe_stacks(stack_name:"woollyams-test")  
=> nil
```

This makes dealing with large numbers of stacks inefficient:

```
Aws> Aws.cloudformation.resource.stacks.map(&:outputs); nil
[Aws::CloudFormation::Client 200 0.886849 0 retries] describe_stacks()  
[Aws::CloudFormation::Client 200 0.11976 0 retries] describe_stacks(stack_name:"woollyams-test")  
[Aws::CloudFormation::Client 200 0.061462 0 retries] describe_stacks(stack_name:"tlap-woollyams")  
[Aws::CloudFormation::Client 200 0.063231 0 retries] describe_stacks(stack_name:"cp-agentadmin-in-e2e-shipper-internal")  
[Aws::CloudFormation::Client 200 0.061777 0 retries] describe_stacks(stack_name:"cp-agentadmin-in-e2e-shipper-public")  
[Aws::CloudFormation::Client 200 0.061863 0 retries] describe_stacks(stack_name:"property-updates-notifier-dev-mwang")  
[Aws::CloudFormation::Client 200 0.062491 0 retries] describe_stacks(stack_name:"inbox-resources")  
[Aws::CloudFormation::Client 200 0.071136 0 retries] describe_stacks(stack_name:"property-cl-ingestor")  
[Aws::CloudFormation::Client 200 0.060383 0 retries] describe_stacks(stack_name:"mercury-pseeker-etl-test")  
[Aws::CloudFormation::Client 200 0.068409 0 retries] describe_stacks(stack_name:"dean-toy-cbde")  
[Aws::CloudFormation::Client 200 0.062357 0 retries] describe_stacks(stack_name:"log-cabin-test")  
[Aws::CloudFormation::Client 200 0.059554 0 retries] describe_stacks(stack_name:"property-data-manager")  
[Aws::CloudFormation::Client 200 0.081533 0 retries] describe_stacks(stack_name:"kate-demo")  
[Aws::CloudFormation::Client 200 1.280546 2 retries] describe_stacks(stack_name:"luke-autoscaling-rolling-update")  
[Aws::CloudFormation::Client 200 0.073418 0 retries] describe_stacks(stack_name:"luke-splunk-debugging-bigger-instance")  
[Aws::CloudFormation::Client 200 0.460789 1 retries] describe_stacks(stack_name:"resi-bamboo-centos-agent")  
[Aws::CloudFormation::Client 200 0.500204 1 retries] describe_stacks(stack_name:"bamboo-agents-docker-1-11")  
[Aws::CloudFormation::Client 200 0.508596 1 retries] describe_stacks(stack_name:"commplat-jobs-dev-wtyan")  
[Aws::CloudFormation::Client 200 0.543895 1 retries] describe_stacks(stack_name:"p4ep-perf")  
[Aws::CloudFormation::Client 200 0.502598 1 retries] describe_stacks(stack_name:"mercury-pseeker-etl")  
[Aws::CloudFormation::Client 200 0.515667 1 retries] describe_stacks(stack_name:"security-monkey-read-only-policy")  
[Aws::CloudFormation::Client 200 0.066568 0 retries] describe_stacks(stack_name:"property-cl-suppression")  
[Aws::CloudFormation::Client 200 0.506252 1 retries] describe_stacks(stack_name:"point-to-cadastre")  
[Aws::CloudFormation::Client 200 0.087089 0 retries] describe_stacks(stack_name:"cpdev-buildkite-agents")  
[Aws::CloudFormation::Client 200 0.480356 1 retries] describe_stacks(stack_name:"point-to-cadastre-rds")  
...etc...
=> nil
```

**Comment by mdub** [comment]
Specifying an appropriate `path` in `aws-sdk-core/apis/cloudformation/2010-05-15/resources-1.json` seems to do the trick:

```diff
diff --git a/aws-sdk-core/apis/cloudformation/2010-05-15/resources-1.json b/aws-sdk-core/apis/cloudformation/2010-05-15/resources-1.json
index 5fa2e6f..ffacfb1 100644
--- a/aws-sdk-core/apis/cloudformation/2010-05-15/resources-1.json
+++ b/aws-sdk-core/apis/cloudformation/2010-05-15/resources-1.json
@@ -32,15 +32,16 @@
     "hasMany": {
       "Stacks": {
         "request": { "operation": "DescribeStacks" },
         "resource": {
           "type": "Stack",
           "identifiers": [
             { "target": "Name", "source": "response", "path": "Stacks[].StackName" }
-          ]
+          ],
+          "path": "Stacks[]"
         }
       }
     }
   },
   "resources": {
     "Event": {
       "identifiers": [
```

Is it as simple as that, or is there a good reason those resources don't have `path` defined?

**Comment by awood45** [comment]
Thanks for the research on this. What you're doing appears on the surface to be correct, but I have to admit I'm not super familiar with CloudFormation details, so I'll review this with my team to make sure we're not missing something. It's very possible that this was just a design oversight.

**Comment by mdub** [comment]
Any further thoughts on this?

Would it be helpful if I put a pull-request together, adding the relevant `path` declarations?

**Comment by awood45** [comment]
I've merged the PR (thank you!) and it will go out with the next release. I'll push it upstream so that other SDKs can pick it up as well!

**Comment by mdub** [comment]
Fixed in #1311 - thanks @awood45.  But unfortunately it was reintroduced in 990c5be :-(

I'll open another issue.
